### PR TITLE
Let mechs open firelock airlocks

### PR DESCRIPTION
**Pull request by MacHac**
### Intent of your Pull Request
You can open firelock-based airlocks while in a mech.

#### Changelog
:cl:  
bugfix: Mechs can open atmospheric firelocks.
/:cl:

### CODE REVIEW

**Comment by Marmio64** [comment]
speed merge please

**Review by SomeguyManperson — APPROVED** [review]

**Comment by Hopekz** [comment]
Not completely against this but thought this was a feature.
Even if it isn't a feature I like mechs not being able to do this so I guess its an accidental feature.

Mechs have never been able to open fully closed firelocks from what I can tell would be nice if this was a feature to specific mechs like the firefighting one

**Comment by MacHac** [comment]
Everything's a feature until someone decides the game is better off without it.  In practice, mech pilots just punch the firelocks down, which takes a while and causes problems for everyone not in a spacewind-proof pressurized mech.

It's just such an arbitrary restriction that even if I closed this right now, it would still seem like an oversight and get patched out eventually.  If you want to nerf mechs or buff the firefighter, there are ways to do that besides undocumented and unintuitive restrictions.

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -70,14 +70,19 @@
...
+	if(panel_open || operating || welded || (stat & NOPOWER))
 		return
 	if(ismob(AM))
 		var/mob/user = AM
-		if(density && !welded && !operating && !(stat & NOPOWER) && (!density || allow_hand_open(user)))
+		if(allow_hand_open(user))
 			add_fingerprint(user)
 			open()
 			return TRUE
+	if(istype(AM, /obj/mecha))
```
Comment (alexkar598) [inline code/game/machinery/doors/firedoor.dm]:
```suggestion
	if(ismecha(AM))
```

**Review by alexkar598 — DISMISSED** [review]

**Review by Hopekz — APPROVED** [review]
Alex's changes have been addressed.
Code checks out.

Reception seems good. Merging.

**Pull request merged**